orchestrator/cli/generator/generator/workflow.py

```diff
@@ -55,10 +55,18 @@ def add_optional_nso_config(config: dict) -> dict:
 
 def insert_lazy_workflow_instances(environment: Environment, config: dict, writer: Callable) -> None:
     template = environment.get_template("lazy_workflow_instance.j2")
+    variable = config.get("variable", "")
+    # product_blocks = config.get("product_blocks", [])
     content = template.render(product=config)
 
     path = settings.FOLDER_PREFIX / settings.WORKFLOWS_PATH / Path("__init__.py")
-    writer(path, content, append=True)
+    if not path.exists():
+        writer(path, "from orchestrator.workflows import LazyWorkflowInstance\n\n")
+    with open(path, "r") as fp:
+        if f"workflows.{variable}.create_{variable}" in fp.read():
+            logger.warning("not re-adding lazy workflows", product=variable)
+        else:
+            writer(path, content, append=True)
```

Comment:
This will probably work but it's a bit cleaner to close the read handle before writing

Reply:
It doesn't hurt to have the file open for read and pointing to end of file, but I agree that it would be better to close the file when we do not want to read from it anymore, if only to release an open file slot. Added a close() before calling the writer().